cmd/kubeadm/app/phases/upgrade/health.go

```diff
@@ -59,49 +66,144 @@ func (c *healthCheck) Name() string {
 // - all control-plane Nodes are Ready
 // - (if self-hosted) that there are DaemonSets with at least one Pod for all control plane components
 // - (if static pod-hosted) that all required Static Pod manifests exist on disk
-func CheckClusterHealth(client clientset.Interface, ignoreChecksErrors sets.String) error {
-	fmt.Println("[upgrade] Making sure the cluster is healthy:")
+func CheckClusterHealth(client clientset.Interface, cfg *kubeadmapi.ClusterConfiguration, ignoreChecksErrors sets.String) error {
+	fmt.Println("[upgrade] Running cluster health checks")
 
 	healthChecks := []preflight.Checker{
 		&healthCheck{
-			name:   "APIServerHealth",
+			name:   "CreateJob",
 			client: client,
-			f:      apiServerHealthy,
+			cfg:    cfg,
+			f:      createJob,
 		},
 		&healthCheck{
 			name:   "ControlPlaneNodesReady",
 			client: client,
 			f:      controlPlaneNodesReady,
 		},
-		// TODO: Add a check for ComponentStatuses here?
+		&healthCheck{
+			name:   "StaticPodManifest",
+			client: client,
+			cfg:    cfg,
+			f:      staticPodManifestHealth,
+		},
 	}
 
-	healthChecks = append(healthChecks, &healthCheck{
-		name:   "StaticPodManifest",
-		client: client,
-		f:      staticPodManifestHealth,
-	})
-
 	return preflight.RunChecks(healthChecks, os.Stderr, ignoreChecksErrors)
 }
 
-// apiServerHealthy checks whether the API server's /healthz endpoint is healthy
-func apiServerHealthy(client clientset.Interface) error {
-	healthStatus := 0
+// CreateJob is a check that verifies that a Job can be created in the cluster
+func createJob(client clientset.Interface, cfg *kubeadmapi.ClusterConfiguration) (lastError error) {
+	jobName := "upgrade-health-check"
+	ns := metav1.NamespaceSystem
+	timeoutSeconds := 15
```

Comment:
These three can all be consts. The last one can be:
`timeout = 15 * time.Second`

Reply:
updated.